pimcore/models/Asset.php

```diff
@@ -987,50 +992,55 @@ public function delete()
 
         \Pimcore::getEventDispatcher()->dispatch(AssetEvents::PRE_DELETE, new AssetEvent($this));
 
-        $this->closeStream();
+        try {
+            $this->closeStream();
 
-        // remove childs
-        if ($this->hasChildren()) {
-            foreach ($this->getChildren() as $child) {
-                $child->delete();
+            // remove childs
+            if ($this->hasChildren()) {
+                foreach ($this->getChildren() as $child) {
+                    $child->delete();
+                }
             }
-        }
 
-        // remove file on filesystem
-        $fullPath = $this->getRealFullPath();
-        if ($fullPath != '/..' && !strpos($fullPath, '/../') && $this->getKey() !== '.' && $this->getKey() !== '..') {
-            $this->deletePhysicalFile();
-        }
+            // remove file on filesystem
+            $fullPath = $this->getRealFullPath();
+            if ($fullPath != '/..' && !strpos($fullPath, '/../') && $this->getKey() !== '.' && $this->getKey() !== '..') {
+                $this->deletePhysicalFile();
+            }
 
-        $versions = $this->getVersions();
-        foreach ($versions as $version) {
-            $version->delete();
-        }
+            $versions = $this->getVersions();
+            foreach ($versions as $version) {
+                $version->delete();
+            }
 
-        // remove permissions
-        $this->getDao()->deleteAllPermissions();
+            // remove permissions
+            $this->getDao()->deleteAllPermissions();
 
-        // remove all properties
-        $this->getDao()->deleteAllProperties();
+            // remove all properties
+            $this->getDao()->deleteAllProperties();
 
-        // remove all metadata
-        $this->getDao()->deleteAllMetadata();
+            // remove all metadata
+            $this->getDao()->deleteAllMetadata();
 
-        // remove all tasks
-        $this->getDao()->deleteAllTasks();
+            // remove all tasks
+            $this->getDao()->deleteAllTasks();
 
-        // remove dependencies
-        $d = $this->getDependencies();
-        $d->cleanAllForElement($this);
+            // remove dependencies
+            $d = $this->getDependencies();
+            $d->cleanAllForElement($this);
 
-        // remove from resource
-        $this->getDao()->delete();
+            // remove from resource
+            $this->getDao()->delete();
 
-        // empty object cache
-        $this->clearDependentCache();
+            // empty object cache
+            $this->clearDependentCache();
 
-        //set object to registry
-        \Pimcore\Cache\Runtime::set('asset_' . $this->getId(), null);
+            //set object to registry
```

Comment:
Misleading doc seems to be there since ages? (nulls it out instead of setting it. Besides that, in this context we are talking about assets ... also see the similar stuff below). Can we correct this together with this PR? :-)

Reply:
Done :-)